src/grpc.ts

```diff
@@ -389,6 +432,85 @@ export class GrpcClient {
     return stub;
   }
 
+  /**
+   * Detect mTLS client certificate based on logic described in
+   * https://google.aip.dev/auth/4114.
+   *
+   * @param {object} [options] - The configuration object.
+   * @returns {Promise} Resolves array of strings representing cert and key.
+   */
+  async _detectClientCertificate(opts?: ClientOptions) {
+    const certRegex =
+      /(?<cert>-----BEGIN CERTIFICATE-----[\s\S]*-----END CERTIFICATE-----)/;
```

Comment:
nit: using just `.` with `/s` modifier is likely more readable. Also, `?` to make sure we grab just one certificate if for any random reason there are two of them in the file.

```
/(?<cert>-----BEGIN CERTIFICATE-----.*?-----END CERTIFICATE-----)/s;  // <-- note the /s modifier to allow . match \n
```